server/src/main/java/org/elasticsearch/indices/IndicesWriteLoadStatsCollector.java

```diff
@@ -0,0 +1,238 @@
+                )
+            );
+            long totalIndexingTime = Math.addExact(indexShard.getTotalIndexingTimeInNanos(), indexShard.getTotalDeleteTimeInNanos());
+            long totalMergeTimeInMillis = indexShard.getTotalMergeTimeInMillis();
+            long totalRefreshTimeInNanos = indexShard.getTotalRefreshTimeInNanos();
+            shardWriteLoadHistogram.recordSample(totalIndexingTime, totalMergeTimeInMillis, totalRefreshTimeInNanos);
+        }
+    }
+
+    List<IndexShard> getDataStreamsWriteIndicesShards() {
+        // Wait until the node has access to the cluster state
+        if (clusterStateReady.get() == false) {
+            return Collections.emptyList();
+        }
+
+        return StreamSupport.stream(indicesService.spliterator(), false)
+            .filter(indexService -> isDataStreamWriteIndex(indexService.index()))
+            .flatMap(indexService -> StreamSupport.stream(indexService.spliterator(), false))
+            .toList();
+    }
+
+    private void cleanRolledOverIndices() {
+        histograms.entrySet().removeIf(entry -> {
+            final var shardId = entry.getKey();
+            return isDataStreamWriteIndex(shardId.getIndex()) == false;
+        });
+    }
+
+    protected Iterable<IndexShard> getShardsForIndex(Index index) {
+        // If the shards are allocated somewhere else, just return an empty list
+        if (indicesService.hasIndex(index) == false) {
+            return Collections.emptyList();
+        }
+
+        return indicesService.indexServiceSafe(index);
+    }
+
+    @Nullable
+    String getParentDataStreamName(String indexName) {
+        IndexAbstraction indexAbstraction = clusterService.state().getMetadata().getIndicesLookup().get(indexName);
+        IndexAbstraction.DataStream parentDataStream = indexAbstraction.getParentDataStream();
+        return parentDataStream != null ? parentDataStream.getName() : null;
+    }
+
+    private boolean isDataStreamWriteIndex(Index index) {
+        final var indicesLookup = clusterService.state().metadata().getIndicesLookup();
+        final var indexAbstraction = indicesLookup.get(index.getName());
+        if (indexAbstraction == null) {
+            return false;
+        }
+
+        final var parentDataStream = indexAbstraction.getParentDataStream();
+        return parentDataStream != null && index.equals(parentDataStream.getWriteIndex());
+    }
+
+    public List<ShardWriteLoadDistribution> getWriteLoadDistributionAndReset() {
+        final var shardWriteLoadDistributions = histograms.values()
+            .stream()
+            .map(ShardWriteLoadHistogram::getWriteLoadDistributionAndReset)
+            .toList();
+
+        cleanRolledOverIndices();
+
+        return shardWriteLoadDistributions;
+    }
+
+    public List<ShardWriteLoadDistribution> getShardLoadDistributions() {
+        return histograms.values().stream().map(ShardWriteLoadHistogram::getWriteLoadDistribution).toList();
+    }
```

Comment:
This method looks unused and once removed, we can make `getWriteLoadDistribution` private and then remove the synchronized inside it.